controllers/apps/transformer_component_tls.go

```diff
@@ -141,13 +142,21 @@ func buildTLSCert(ctx context.Context, cli client.Reader, synthesizedComp compon
 		return fmt.Errorf("issuer shouldn't be nil when tls enabled")
 	}
 
+	var getSecretErr error
 	switch tls.Issuer.Name {
 	case appsv1alpha1.IssuerUserProvided:
 		if err := plan.CheckTLSSecretRef(ctx, cli, synthesizedComp.Namespace, tls.Issuer.SecretRef); err != nil {
 			return err
 		}
 	case appsv1alpha1.IssuerKubeBlocks:
-		// TODO: (good-first-issue) should check if the secret exists, skip if already exists
+		secretName := plan.GenerateTLSSecretName(synthesizedComp.ClusterName, synthesizedComp.Name)
+		preSecret := &corev1.Secret{}
+		if getSecretErr = cli.Get(ctx, types.NamespacedName{Namespace: synthesizedComp.Namespace, Name: secretName}, preSecret); !errors.IsNotFound(getSecretErr) {
```

Comment:
rename `getSecretErr` to `err` and use short assignment is enough: `if err := ...`